src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp

```diff
@@ -95,16 +93,18 @@ void BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(
   bool const use_deprecated_nearest_query_algorithm =
       which == NearestQueryAlgorithm::PriorityQueueBased_Deprecated;
 
-  auto const n_queries = queries.extent(0);
+  using Access = Traits::Access<Predicates, Traits::PredicatesTag>;
+  auto const n_queries = Access::size(predicates);
 
   Kokkos::Profiling::pushRegion("ArborX:BVH:sort_queries");
 
   auto const permute =
       Details::BatchedQueries<DeviceType>::sortQueriesAlongZOrderCurve(
-          bvh.bounds(), queries);
+          bvh.bounds(), predicates);
 
-  queries =
-      Details::BatchedQueries<DeviceType>::applyPermutation(permute, queries);
+  // FIXME  readability!  queries is a sorted copy of the predicates
```

Comment:
What is the readability issue here?

Reply:
Function argument is `predicates` and it is some unspecified type that is only required to have a valid matching specialization of the access traits and bam next line I create `queries` which is a Kokkos view and keep going as before.